### PR TITLE
Log proxy request success

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Logging some metrics from our list

## Proposal

Implementing proxy request success Counter metric. We'll be able to calculate the error and success rate using this and the error counter

## Test Plan
1. Ran validator locally with `./build_and_redeploy.sh --port-forward --clean`
2. Built the binaries with `cargo build --features benchmark`
3. Ran a benchmark against it with `./linera benchmark`
4. Port forwarded the Prometheus UI `kubectl port-forward prometheus-linera-core-kube-prometheu-prometheus-0 9090`
5. Went to http://127.0.0.1:9090/graph to see the metric properly logged:

### CODE REVIEW

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1166** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1166" target="_blank">Graphite</a> 
    * **PR #1169** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1169" target="_blank">Graphite</a> 
      * **PR #1170** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1170" target="_blank">Graphite</a> 
        * **PR #1171** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171" target="_blank">Graphite</a>  👈
          * **PR #1172** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171?utm_source=stack-comment).

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1166** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1166" target="_blank">Graphite</a> 
    * **PR #1169** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1169" target="_blank">Graphite</a> 
      * **PR #1170** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1170" target="_blank">Graphite</a> 
        * **PR #1171** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171" target="_blank">Graphite</a>  👈
          * **PR #1172** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171?utm_source=stack-comment).

**Review by MathieuDutSik — COMMENTED** [review]
This is a sequence of PR introducing metrics.
For the proof of work in the section `Test Plan` a screenshot is provided with the mention "Ran validator locally, ran a benchmark against it, and saw metric properly logged".

I find that not enough to be able as a user to reproduce and say reproduce the screenshot. Could you provide step by step instructions that allow a user to reproduce that screenshot.

**Review by afck — APPROVED** [review]

File: linera-service/src/grpc_proxy.rs [hunk 1]
```diff
@@ -230,7 +240,15 @@ impl ValidatorNode for GrpcProxy {
...
-        client.handle_block_proposal(inner).await
+        match client.handle_block_proposal(inner).await {
+            Ok(chain_info_result) => {
+                PROXY_REQUEST_SUCCESS
+                    .with_label_values(&["handle_block_proposal"])
+                    .inc();
+                Ok(chain_info_result)
+            }
+            Err(status) => Err(status),
+        }
```
[the remark this answers is not in the document]
Comment (andresilva91) [reply linera-service/src/grpc_proxy.rs:251]: Ah, yes, good point

**Comment by andresilva91** [comment]
> This is a sequence of PR introducing metrics.
> For the proof of work in the section `Test Plan` a screenshot is provided with the mention "Ran validator locally, ran a benchmark against it, and saw metric properly logged".
> 
> I find that not enough to be able as a user to reproduce and say reproduce the screenshot. Could you provide step by step instructions that allow a user to reproduce that screenshot.

`Ran validator locally, ran a benchmark against it, and saw metric properly logged` was my attempt at a step by step for reproducing it, but you're right that someone with no context would have no idea what that means :)
I'll add the commands I ran to make it clearer

File: linera-service/src/grpc_proxy.rs [hunk 2]
```diff
@@ -257,7 +291,15 @@ impl ValidatorNode for GrpcProxy {
...
+                PROXY_REQUEST_SUCCESS
+                    .with_label_values(&["handle_chain_info_query"])
+                    .inc();
+                Ok(chain_info_result)
+            }
+            Err(status) => Err(status),
+        }
     }
 
     #[instrument(skip_all, err(Display))]
```
Comment (andresilva91) [inline linera-service/src/grpc_proxy.rs:305]: <i class='graphite__hidden'>[Re: lines 306 to 306]<br/></i>

I'm right in assuming we don't want to log metrics on this one, right?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (andresilva91) [reply linera-service/src/grpc_proxy.rs:300]: Synched on Slack, will add this later

**Comment by andresilva91** [comment]
### Merge activity

* **Oct 30, 3:05 PM**: @andresilva91 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171).

**Pull request merged**